script/openqa-load-templates

```diff
@@ -120,16 +120,34 @@ sub post_entry ($table, $entry) {
     if ($table eq 'JobGroups') {
         # Create the group first
         my $job_groups_url = $url->clone->path($options{apibase} . '/job_groups');
-        my $res = $client->post($job_groups_url, form => {name => $entry->{group_name}})->res;
-        print_error $res unless $res->is_success;
-
+        my $existing_group_res = $client->get($job_groups_url, form => {name => $entry->{group_name}})->res;
+        my $group_exists = 0;
+        if ($existing_group_res->is_success) {
+            my $json = $existing_group_res->json;
+            for my $group (@$json) {
+                if ($group->{name} eq $entry->{group_name}) {
+                    $group_exists = 1
+                      if $options{update}
+                      or die
+                      "Job group '$entry->{group_name}' already exists. Use --update to modify existing entries.";
+                }
+            }
+        }
```

Comment:
```suggestion
        my $existing_group_res = $client->get($job_groups_url, form => {name => $entry->{group_name}})->res;
        print_error $res unless $existing_group_res->is_success;
        my $group_exists = 0;
            for my $group (@$existing_group_res->json) {
                next unless $group->{name} eq $entry->{group_name};
                $group_exists = 1
                    if $options{update}
                    or die
                    "Job group '$entry->{group_name}' already exists. Use --update to modify existing entries.";
            }
```

Reply:
I'm not sure why your suggestion `for my $group (@$existing_group_res->json) {` isn't working here, but i've included your changes and perl insists on me using the intermediate variable `my $json = $existing_group_res->json` for the loop, otherwise the tests fail with `Not a HASH reference at .../openQA/script/openqa-load-templates` :thinking: